subgraphs/enzyme-core/mappings/SingleAssetDepositQueueLib.ts

```diff
@@ -0,0 +1,134 @@
+import { arrayDiff, arrayUnique, toBigDecimal } from '@enzymefinance/subgraph-utils';
+import { Address } from '@graphprotocol/graph-ts';
+import {
+  ensureSingleAssetDepositQueue,
+  ensureSingleAssetDepositQueueRequest,
+} from '../entities/SingleAssetDepositQueue';
+import { useVault } from '../entities/Vault';
+import {
+  Initialized,
+  ManagerAdded,
+  ManagerRemoved,
+  Deposited,
+  DepositRequestAdded,
+  RequestBypassed,
+  RequestCanceled,
+  Shutdown,
+  MinDepositAssetAmountSet,
+  MinRequestTimeSet,
+  DepositorAllowlistIdSet,
+} from '../generated/contracts/SingleAssetDepositQueueLibEvents';
+import { createAssetAmount } from '../entities/AssetAmount';
+import { ensureAsset, isAsset } from '../entities/Asset';
+import { ensureAccount } from '../entities/Account';
+
+export function handleInitialized(event: Initialized): void {
+  let depositQueue = ensureSingleAssetDepositQueue(event.address, event);
+  depositQueue.vault = useVault(event.params.vaultProxy.toHex()).id;
+  if (isAsset(event.params.depositAsset)) {
+    let depositAsset = ensureAsset(event.params.depositAsset);
+    depositQueue.depositAsset = depositAsset.id;
+  }
+  depositQueue.save();
+}
+
+export function handleManagerAdded(event: ManagerAdded): void {
+  let manager = ensureAccount(event.params.user, event);
+
+  let depositQueue = ensureSingleAssetDepositQueue(event.address, event);
+  depositQueue.managers = arrayUnique(depositQueue.managers.concat([manager.id]));
+  depositQueue.save();
+}
+
+export function handleManagerRemoved(event: ManagerRemoved): void {
+  let manager = ensureAccount(event.params.user, event);
+
+  let depositQueue = ensureSingleAssetDepositQueue(event.address, event);
+  depositQueue.managers = arrayDiff(depositQueue.managers, [manager.id]);
+  depositQueue.save();
+}
+
+export function handleDeposited(event: Deposited): void {
+  let queue = ensureSingleAssetDepositQueue(event.address, event);
+
+  let request = ensureSingleAssetDepositQueueRequest(queue, event.params.id, event);
+  request.deposited = true;
+  request.depositedAt = event.block.timestamp.toI32();
+  request.sharesAmount = toBigDecimal(event.params.sharesAmountReceived, 18);
+  request.save();
+}
+
+export function handleDepositRequestAdded(event: DepositRequestAdded): void {
+  let queue = ensureSingleAssetDepositQueue(event.address, event);
+
+  let depositAssetAddress = queue.depositAsset;
+  if (depositAssetAddress == null) {
+    return;
+  }
+
+  let request = ensureSingleAssetDepositQueueRequest(queue, event.params.id, event);
+  request.createdAt = event.block.timestamp.toI32();
+
+  let depositAsset = ensureAsset(
+    Address.fromString(
+      // as string added because subgraph is not smart enough to checkout that depositAssetAddress is not null, because of the condition above
+      depositAssetAddress as string,
+    ),
+  );
```

Comment:
This will probably be simpler if you store both the asset address and the link to the Asset entity (see above).